### ISSUE TITLE
Do not use unchecked multiply

### ISSUE THREAD

**Issue by linkmauve** [tafia/quick-xml#101]
```
thread '<unnamed>' panicked at 'attempt to multiply with overflow', /home/linkmauve/data/cargo/registry/src/github.com-1ecc6299db9ec823/quick-xml-0.10.0/src/escape.rs:166:9
```

Here is the testcase generated by cargo-fuzz:
```
\x00\x00<\x00\x00\x0a>&#44444444401?#\x0a413518#\x0a\x0a\x0a;<:<)(<:\x0a\x0a\x0a\x0a;<:\x0a\x0a<:\x0a\x0a\x0a\x0a\x0a<\x00*\x00\x00\x00\x00
```

The `&#44444444401` part seems to be the issue, multiplying that by ten does not fit in an `u32`, causing the panic.

This bug has been found while fuzzing [xmpp-parsers](https://hg.linkmauve.fr/xmpp-parsers) which uses [minidom](https://gitlab.com/lumi/minidom-rs), you would benefit a lot from fuzzing quick-xml itself.

**Comment by tafia** [comment]
Nice.

I [already fuzzed quick-xml](https://github.com/tafia/quick-xml/blob/1a40e2ea01e88e7e153bc817d915c2658b91320c/tests/test.rs#L332-L345). 
Either 
- the error is due to recent changes
- I didn't wait long enough for this one.
- i didn't test the right part of the code

Thanks!!

**Comment by tafia** [comment]
Published version 0.10.1
